src/mode.js

```diff
@@ -106,15 +100,13 @@ function getRtvVersion(win) {
  * bypassed via #validate=0.
  * Note that AMP_DEV_MODE flag is used for testing purposes.
  * @param {!Window} win
+ * @param {!JsonObject=} opt_hashParams
  * @return {boolean}
  */
-export function isModeDevelopment(win) {
-  const hashQuery = parseQueryString(
-    win.location['originalHash'] || win.location.hash
-  );
+export function isModeDevelopment(win, opt_hashParams) {
   return !!(
     ['1', 'actions', 'amp', 'amp4ads', 'amp4email'].includes(
-      hashQuery['development']
+      (opt_hashParams || getHashParams(win))['development']
     ) || win.AMP_DEV_MODE
```

Comment:
nit: i'd prefer extracting some of these to variables. imo we are doing a bit too much in a single expression.

```js
const devParam = (opt_hashParams || getHashParams(win))['development'];
const devModes = ['1', 'actions', 'amp', 'amp4ads', 'amp4email'];
return devModes.includes(devParam) || win.AMP_DEV_MODE;
```